src/Platforms.jl

```diff
@@ -207,3 +207,55 @@ function Pkg.Artifacts.pack_platform!(meta::Dict, p::ExtendedPlatform)
     # override the default ones
     Artifacts.pack_platform!(meta, base_platform(p))
 end
+
+# Get the list of CPU features for the host system querying libLLVM
+function get_cpu_features()
+    if VERSION < v"1.4"
+        error("Cannot automatically detect the features of the CPU with this version of Julia!")
+    end
```

Comment:
If we go by this route, all the microarchitecture stuff will work only with Julia v1.4+, as this requires libllvm v8+.  Not too bad since the alternative would be to have some code in Base to do the same, but that would require Julia v1.6 or more.